### PR TITLE
Add k8s client and storage grpc server options

### PR DESCRIPTION
**Pull request by aguerra**
This change is [Reviewable](https://reviewable.io/reviews/luizalabs/teresa-api/202)

### CODE REVIEW

File: pkg/server/k8s/k8s.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+func New(conf *Config) (Client, error) {
+	validServiceTypes := map[api.ServiceType]bool{
+		api.ServiceTypeLoadBalancer: true,
+		api.ServiceTypeNodePort:     true,
+		api.ServiceTypeClusterIP:    true,
+	}
+	serviceType := api.ServiceType(conf.DefaultServiceType)
+	if _, ok := validServiceTypes[serviceType]; !ok {
+		return nil, fmt.Errorf(
```
Comment (drgarcia1986) [inline pkg/server/k8s/k8s.go]: maybe a known error (I mean `ErrInvalidServiceType` f.ex.)

File: pkg/server/storage/s3.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	Secret           string
+	Region           string
+	Bucket           string
+	Endpoint         string
+	DisableSSL       bool
+	S3ForcePathStyle bool
+}
+
+func (s *S3) GetK8sSecretName() string {
+	return "s3-storage"
```
Comment (drgarcia1986) [inline pkg/server/storage/s3.go]: Are you thinking to use k8s secrets to handle aws credentials ?
(to tell you true, I don't understand why we need secrets for s3 integration)

File: pkg/server/k8s/k8s.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	Secrets(namespace string) unversioned.SecretsInterface
+	HorizontalPodAutoscalers(namespace string) unversioned.HorizontalPodAutoscalerInterface
+}
+
+func New(conf *Config) (Client, error) {
+	validServiceTypes := map[api.ServiceType]bool{
+		api.ServiceTypeLoadBalancer: true,
+		api.ServiceTypeNodePort:     true,
+		api.ServiceTypeClusterIP:    true,
+	}
```
Comment (aguerra) [inline pkg/server/k8s/k8s.go]: Sure, copi & pasta effect.

File: pkg/server/k8s/k8s.go (hunk 1 above)
Comment (aguerra) [inline pkg/server/k8s/k8s.go]: Sure, copi & pasta effect.

File: pkg/server/k8s/k8s_test.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+	}
+}
+
+func TestK8sNewInvalidDefaultServiceType(t *testing.T) {
+	conf := &Config{DefaultServiceType: "Invalid type"}
+
+	if _, err := New(conf); err == nil {
+		t.Error("expected error, got nil")
+	}
+}
```
Comment (aguerra) [inline pkg/server/k8s/k8s_test.go]: :+1:

File: pkg/server/storage/storage.go [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+	AwsRegion           string      `envconfig:"aws_region"`
+	AwsBucket           string      `envconfig:"aws_bucket"`
+	AwsEndpoint         string      `envconfig:"aws_endpoint" default:""`
+	AwsDisableSSL       bool        `envconfig:"aws_disable_ssl" default:"false"`
+	AwsS3ForcePathStyle bool        `envconfig:"aws_s3_force_path_style" default:"false"`
+}
+
+type Storage interface {
+	GetK8sSecretName() string
+	GetAccessData() map[string][]byte
```
Comment (aguerra) [inline pkg/server/storage/storage.go]: Sure, copi & pasta effect.

File: pkg/server/storage/storage_test.go [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+	}
+}
+
+func TestNewStorageInvalidType(t *testing.T) {
+	conf := &Config{Type: "Invalid type"}
+
+	if _, err := New(conf); err == nil {
+		t.Error("expected error, got nil")
+	}
+}
```
Comment (aguerra) [inline pkg/server/storage/storage_test.go]: :+1:

File: pkg/server/storage/s3.go (hunk 2 above)
Comment (aguerra) [inline pkg/server/storage/s3.go]: The builder and runner need it.

File: pkg/server/storage/s3_test.go [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+	if string(ad["builder-bucket"]) != conf.AwsBucket {
+		t.Errorf("expected %s, got %s", conf.AwsBucket, ad["builder-bucket"])
+	}
+	if string(ad["accesskey"]) != conf.AwsKey {
+		t.Errorf("expected %s, got %s", conf.AwsKey, ad["accesskey"])
+	}
+	if string(ad["secretkey"]) != conf.AwsSecret {
+		t.Errorf("expected %s, got %s", conf.AwsSecret, ad["secretkey"])
+	}
+}
```
Comment (drgarcia1986) [inline pkg/server/storage/s3_test.go]: maybe another table test? (follow your heart ❤️ )

**Pull request merged**